catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/AsyncFileEntry.java

```diff
@@ -45,7 +44,7 @@
   private boolean directory;
   private long length;
 
-  private final Set<AsyncFileEntry> children = new ConcurrentSkipListSet<>();
+  private final ConcurrentSkipListSet<AsyncFileEntry> children = new ConcurrentSkipListSet<>();
```

Comment:
✏️ Might be worth adding a comment here to indicate why we're not using `Set` to declare `children`.